### PR TITLE
[#1116]

### PR DESCRIPTION
**Pull request by cadecairos**
1. Update events docs page
2. Add trackstart and trackend docs

Lighthouse ticket: https://webmademovies.lighthouseapp.com/projects/63272-popcornjs/tickets/1116-popcorn-specific-events-need-docs#ticket-1116-4

### CODE REVIEW

File: events/_posts/2012-12-12-trackend.md [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+---
+# trackend #
+
+## Purpose ##
+
+`"trackend"` is triggered by Popcorn when any instance of a plugin has its `end` method invoked. `"trackend"` is to be used with the `on` and `off` instance methods.
+
+For example:
+`popcornInstance.on( "trackstart", callbackFunction );`
+`popcornInstance.off( "trackstart", callbacFunction );`
```
Comment (rwaldron) [inline events/_posts/2012-12-12-trackend.md]: "trackstart" => "trackend"

**Comment by rwaldron** [comment]
We should add "canplayall", which is a Popcorn specific event that is bound to the _first_ time "canplaythrough" is fired for a media element.

File: events/_posts/2012-12-12-canplayall.md [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+* `popcornInstance.on( "canplayall", callbackFunction );`
+* `popcornInstance.off( "canplayall", callbackFunction );`
+
+## Use Cases ##
+
+`"canplayall"` is useful for determining the point at which a media element is ready to have its `"currentTime"` value changed.
+
+## Examples ##
+
+[Live demo using trackend](http://jsfiddle.net/popcornjs/PTTrq/)
```
Comment (mjschranz) [inline events/_posts/2012-12-12-canplayall.md]: Fix this to be <pre>Live demo using canplayall</pre>

File: events/_posts/2012-12-12-canplayall.md [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+## Related ##
+
+[media-events](#media-events)
```
Comment (mjschranz) [inline events/_posts/2012-12-12-canplayall.md]: Can the link text here be Media Events instead?

File: events/_posts/2012-12-12-mediaevents.md [hunk 4]
```diff
@@ -0,0 +1,163 @@
...
+    <td>The metadata has loaded or changed, indicating a change in duration of
+      the media.&nbsp; This is sent, for example, when the media has loaded enough
+      that the duration is known.</td>
+  </tr>
+  <tr>
+    <td><code>emptied</code>
+    </td>
+    <td>The media has become empty; for example, this event is sent if the media
+      has already been loaded (or partially loaded), and the <a title="en/nsIDOMHTMLMediaElement"
+      rel="internal" href="https://developer.mozilla.org/En/XPCOM_Interface_Reference/NsIDOMHTMLMediaElement"><code>load()</code></a>&nbsp;method
```
Comment (rwaldron) [inline events/_posts/2012-12-12-mediaevents.md]: This should be: https://developer.mozilla.org/en-US/docs/DOM/HTMLMediaElement

File: events/_posts/2012-12-12-mediaevents.md [hunk 5]
```diff
@@ -0,0 +1,163 @@
...
+    <td>The media's metadata has finished loading; all attributes now contain
+      as much useful information as they're going to.</td>
+  </tr>
+  <tr>
+    <td><code>loadstart</code>
+    </td>
+    <td>Sent when loading of the media begins.</td>
+  </tr>
+  <tr>
+    <td><code>mozaudioavailable</code>
```
Comment (rwaldron) [inline events/_posts/2012-12-12-mediaevents.md]: Let's avoid promoting anything with a prefix :D

Comment (cadecairos) [inline events/_posts/2012-12-12-mediaevents.md]: +1

File: events/_posts/2012-12-12-mediaevents.md [hunk 6]
```diff
@@ -0,0 +1,156 @@
...
+    <td>Sent when the audio volume changes (both when the volume is set and when
+      the <code>muted</code> attribute is changed).</td>
+  </tr>
+  <tr>
+    <td><code>waiting</code>
+    </td>
+    <td>Sent when the requested operation (such as playback) is delayed pending
+      the completion of another operation (such as a seek).</td>
+  </tr>
+</table>
```
Comment (dseif) [inline events/_posts/2012-12-12-mediaevents.md]: This should be done in the same format as the rest of our docs

File: events/_posts/2012-12-12-trackend.md [hunk 7]
```diff
@@ -0,0 +1,24 @@
+---
+title: trackend
+---
+# trackend #
+
+## Purpose ##
+
+`"trackend"` is triggered by Popcorn when any instance of a plugin has its `end` method invoked. `"trackend"` is to be used with the `on` and `off` instance methods.
+
```
Comment (rwaldron) [inline events/_posts/2012-12-12-trackend.md:9]: The `event` object argument will have a property called `track`, which is a reference to the `TrackEvent` object created by that plugin for the calling Popcorn instance.

File: events/_posts/2012-12-12-trackstart.md [hunk 8]
```diff
@@ -0,0 +1,24 @@
+---
+title: trackstart
+---
+# trackstart #
+
+## Purpose ##
+
+trackstart is triggered by Popcorn when any instance of a plugin has its `start` method invoked. To be used with the `on` and `off` methods.
+
```
Comment (rwaldron) [inline events/_posts/2012-12-12-trackstart.md:9]: The `event` object argument will have a property called `track`, which is a reference to the `TrackEvent` object created by that plugin for the calling Popcorn instance.

**Comment by rwaldron** [comment]
This is awesome. I don't expect you to have time for this as well, but the next events to add are:

`tracksetup`
`trackadded`
`trackteardown`
`trackremoved`
`trackchange`

All of them should include the following:

> The `event` object argument will have a property called `track`, which is a reference to the `TrackEvent` object created by that plugin for the calling Popcorn instance.

**Comment by cadecairos** [comment]
Rick: I'm gonna check if those can be done in a related ticket that's filed, and post this comment there, otherwise I'll file a new one for these events.

**Comment by dseif** [comment]
Yo what's the status of this? Would love to see this get pushed up!

**Comment by rwaldron** [comment]
Good to go!

**Comment by PragTob** [comment]
I think it'd be extremely helpful if this made it in, if it's still accurate =)

**Pull request merged**